yb-voyager/src/srcdb/oracle.go

```diff
@@ -234,6 +234,38 @@ func (ora *Oracle) FilterUnsupportedTables(tableList []*sqlname.SourceName) []*s
 	return tableList
 }
 
+func (ora *Oracle) FilterEmptyTables(tableList []*sqlname.SourceName) []*sqlname.SourceName {
+	nonEmptyTableList := make([]*sqlname.SourceName, 0)
+	for _, tableName := range tableList {
+		query := fmt.Sprintf("SELECT 1 FROM %s WHERE ROWNUM=1", tableName.ObjectName.MinQuoted)
+		if ora.IsNestedTable(tableName) {
+			// query to check empty nested oracle tables
+			query = fmt.Sprintf(`SELECT 1 from dba_segments 
+			where owner = '%s' AND segment_name = '%s' AND segment_type = 'NESTED TABLE'`,
+				tableName.SchemaName.Unquoted, tableName.ObjectName.MinQuoted)
+		}
+
+		if !IsTableEmpty(ora.db, query) {
+			nonEmptyTableList = append(nonEmptyTableList, tableName)
+		} else {
+			log.Infof("Skipping empty table %v", tableName)
+		}
+	}
+	return nonEmptyTableList
+}
+
+func (ora *Oracle) IsNestedTable(tableName *sqlname.SourceName) bool {
+	// sql query to find out if it is oracle nested table
+	chkNestedTblquery := fmt.Sprintf("SELECT 1 FROM ALL_NESTED_TABLES WHERE OWNER = '%s' AND TABLE_NAME = '%s'",
```

Comment:
One should use context of the surrounding code to reduce the identifier size e.g. even if you name this variable as just `query`, one can understand from the surrounding context that the query is for checking if table is nested.